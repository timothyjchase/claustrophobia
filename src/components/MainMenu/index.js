import React from 'react'
import PropTypes from 'prop-types'
import { Dropdown } from 'semantic-ui-react'
import RulesSummaryButton from '../RulesSummaryButton'
import DemonAISummaryButton from '../DemonAISummaryButton'
import GameResultButton from '../GameResultButton'

const MainMenu = ({
  currentState,
  scenarioName,
  resetGame,
  undoLastChange,
  ...divProps
}) => (
  <div {...divProps}>
    <div style={{ float: 'right' }}>
      <Dropdown icon="bars" direction="left" style={{ color: 'white' }}>
        <Dropdown.Menu>
          <Dropdown.Item icon="refresh" text="Restart" onClick={resetGame} />
          <Dropdown.Item icon="undo" text="Undo" onClick={undoLastChange} />
          <Dropdown.Divider />
          <RulesSummaryButton icon="book" text="Rules" as={Dropdown.Item} />
          <DemonAISummaryButton icon="bug" text="Demon AI" as={Dropdown.Item} />
          <Dropdown.Divider />
          <GameResultButton
            icon="thumbs up outline"
            text="Victory"
            as={Dropdown.Item}
            currentState={currentState}
            scenarioName={scenarioName}
            result="Victory"
          />
          <GameResultButton
            icon="thumbs down outline"
            text="Defeat"
            as={Dropdown.Item}
            currentState={currentState}
            scenarioName={scenarioName}
            result="Defeat"
          />
        </Dropdown.Menu>
      </Dropdown>
    </div>
  </div>
)

MainMenu.propTypes = {
  currentState: PropTypes.object.isRequired,
  scenarioName: PropTypes.string, // eslint-disable-line react/require-default-props
  resetGame: PropTypes.func.isRequired,
  undoLastChange: PropTypes.func.isRequired,
}

export default MainMenu
