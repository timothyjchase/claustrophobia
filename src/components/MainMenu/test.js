import React from 'react'
import MainMenu from './index'

describe('<MainMenu />', () => {
  const commonProps = {
    currentState: {},
    scenarioName: 'Test scenario',
    resetGame: jest.fn(),
    undoLastChange: jest.fn(),
  }
  describe('with basic props', () => {
    const component = shallow(<MainMenu {...commonProps} />)
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
