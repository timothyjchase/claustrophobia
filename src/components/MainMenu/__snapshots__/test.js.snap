// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> with basic props should match snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      direction="left"
      icon="bars"
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      style={
        Object {
          "color": "white",
        }
      }
      wrapSelection={true}
    >
      <DropdownMenu>
        <DropdownItem
          icon="refresh"
          onClick={[MockFunction]}
          text="Restart"
        />
        <DropdownItem
          icon="undo"
          onClick={[MockFunction]}
          text="Undo"
        />
        <DropdownDivider />
        <RulesSummaryButton
          as={[Function]}
          icon="book"
          text="Rules"
        />
        <DemonAISummaryButton
          as={[Function]}
          icon="bug"
          text="Demon AI"
        />
        <DropdownDivider />
        <GameOverButton
          as={[Function]}
          currentState={Object {}}
          icon="thumbs up outline"
          result="Victory"
          scenarioName="Test scenario"
          text="Victory"
        />
        <GameOverButton
          as={[Function]}
          currentState={Object {}}
          icon="thumbs down outline"
          result="Defeat"
          scenarioName="Test scenario"
          text="Defeat"
        />
      </DropdownMenu>
    </Dropdown>
  </div>
</div>
`;
